src/coreclr/jit/valuenum.cpp

```diff
@@ -11136,12 +11162,6 @@ void Compiler::fgValueNumberStore(GenTree* store)
 
             valueVNPair.SetBoth(initObjVN);
         }
-        else if (value->TypeGet() == TYP_REF)
-        {
-            // If we have an unsafe IL store of a TYP_REF to a non-ref (typically a TYP_BYREF)
-            // then don't propagate this ValueNumber to the lhs, instead create a new unique VN.
-            valueVNPair.SetBoth(vnStore->VNForExpr(compCurBB, store->TypeGet()));
-        }
```

Comment:
Might be a good idea to see if you can dig up why this was added to verify it won't cause any problems to remove it.